multiraft/multiraft.go

```diff
@@ -388,6 +388,9 @@ type node struct {
 }
 
 func (n *node) registerGroup(groupID uint64) {
+	if groupID == noGroup {
+		panic("must not call registerGroup with noGroup")
+	}
```

Comment:
think of this as an `assert()` and don't worry about it too much. This isn't production code, I just want to know right away if we introduce multiraft flakiness again.